jax/_src/ad_checkpoint.py

```diff
@@ -687,6 +688,9 @@ def remat_dce(used_outputs: list[bool], eqn: core.JaxprEqn
     return used_inputs, new_eqn
 pe.dce_rules[remat_p] = remat_dce
 
+def _has_effects(effects) -> bool:
+  return bool({e for e in effects if not isinstance(e, core.NamedAxisEffect)})
```

Comment:
This could probably be `any(e for f in effects if not isinstance(e, core.NamedAxisEffect))`.